src/engine/QueryPlanner.cpp

```diff
@@ -1768,6 +1794,41 @@ std::vector<QueryPlanner::SubtreePlan> QueryPlanner::createJoinCandidates(
   return candidates;
 }
 
+// _____________________________________________________________________________
+auto QueryPlanner::createSpatialJoin(
+    SubtreePlan a, SubtreePlan b,
+    const std::vector<std::array<ColumnIndex, 2>>& jcs)
+    -> std::optional<SubtreePlan> {
+  using enum QueryExecutionTree::OperationType;
+  const bool aIsSpatialJoin = a._qet->getType() == SPATIAL_JOIN;
+  const bool bIsSpatialJoin = b._qet->getType() == SPATIAL_JOIN;
+  
+  if (!(aIsSpatialJoin || bIsSpatialJoin)
+      || (aIsSpatialJoin && bIsSpatialJoin)) {
+    return std::nullopt;
+  }
```

Comment:
Did you mean
`if (! (aIsSpatialJoin ^ bIsSpatialJoin)) { return std::nullopt;}`
(Exactly one of them has to be).
I find this much easier to read than what you chose.